content/posts/chaos-testing-guide/index.mdx

```diff
@@ -0,0 +1,228 @@
+
+- Netflix Outage on a Saturday Night [Source](https://www.express.co.uk/life-style/science-technology/1516749/Netflix-down-error-watch-online-network-offline-app)
+  - The streaming service was reported down in the US and at least a dozen countries. This led to negative feedback from customers on social media.
+
+
+### Test Strategies and Methodology
+- Failures in production are costly. To help mitigate risk to service health, consider the following strategies and approaches to service testing:
+  - Be proactive vs reactive - introducing error situations and observing and correcting the behavior of the software under test in response to these errors instead of reacting and fixing the issues after encountering them in production environments. We have different types of test suites in place - unit, integration and end-to-end - that help expose bugs in code in a controlled environment.  Through implementation of a chaos engineering strategy, we can help discover potential issues for service degradation that were unknown. We need to understand the systems’ behavior under unpredictable conditions in order to find the areas to harden, and use the performance data points to size the clusters to handle the failures in order to keep the downtime to as minimal as possible.
+
+  - Test the resiliency of a system under turbulent conditions by running tests that are designed to disrupt while monitoring the systems adaptability and performance:
+    - Establish and define your steady state and metrics - understand the behavior and performance under stable conditions  and define the metrics that will be used to evaluate the system’s behavior.  Then decide on acceptable outcomes before injecting chaos. 
+    - Analyze the statuses and metrics of all components during the chaos test runs.
+    - Improve the areas that are not resilient and performant by comparing the key metrics and Service Level Objectives (SLOs) to the stable conditions before the chaos.
+      For example: evaluating the API server latency or application uptime to see if the key performance indicators and service level indicators are still within the acceptable limits.
+
+
+### Best Practices
+Now that we understand the test methodology, let’s take a look at the best practices for an OpenShift cluster.  On that platform there are user applications and cluster workloads that need to be designed for stability and to provide the best user experience possible:
+- Alerts with appropriate severity should get fired
```

Comment:
The sentence construct is a bit confusing in two ways:  if you misinterpret 'fired' as "end the employment of an alert", and if read as, "Fire the alerts that have appropriate severity," I'm not sure it makes sense even in an SRE context?

That is, if it's already an alert, hasn't an alert been fired? What does it mean to "fire an alert" when the alert exists already?

Perhaps "alert" is doing double-duty in this sentence? If so, does this or something similar work?
```suggestion
- System alerts with appropriate severity should trigger a human response.
```

Reply:
Makes sense, tweaked it to "Alerts with appropriate severity". Thanks.